### PR TITLE
AtomicQueue add test coverage

### PR DESCRIPTION
**Pull request by ColinGeukes**
In order to improve the test coverage of the code base of libGDX, this pull request is created. 
In this pull request, the following has been done to the file `utils/AtomicQueue`:

- Added the test coverage 

The code coverage has been improved as follows

| File | Class, % | Method, %  | Line, % |
| ------------- |:-------------:|:-----:|:-----:|
| `utils/AtomicQueue` (old)   | 0% | 0% | 0% |
| `utils/AtomicQueue` (new)      | 100% |   100% | 100% |

The changes to AtomicQueuedocumentation are reverted but an artefact is left as I cannot push spaces to git it does not recognize it as any change.

### CODE REVIEW

File: gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        // We do not have enough space to write another element as the read pointer is there.
+        assertFalse(atomicQueue.put(2));
+    }
+
+    @Test
+    public void PullTest () {
+        AtomicQueue<Integer> atomicQueue = new AtomicQueue<>(3);
+
+        // Add elements, not that the last element can not fit on the array.
```
Comment (intrigus) [inline gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java:23]:
```suggestion
        // Add elements, note that the last element can not fit in the array.
```

File: gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        atomicQueue.put(1);
+        atomicQueue.put(2);
+        atomicQueue.put(3);
+
+        // Poll the two elements on the array.
```
Comment (intrigus) [inline gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java:28]:
```suggestion
        // Poll the two elements in the array.
```

File: gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java (hunk 1 above)
Comment (ColinGeukes) [reply gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java]: Changed.

File: gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java (hunk 2 above)
Comment (ColinGeukes) [reply gdx/test/com/badlogic/gdx/utils/AtomicQueueTest.java]: Changed.

**Pull request merged**